### PR TITLE
refactor(list-to-object): Refactored all endpoints returning lists

### PR DESCRIPTION
**Pull request by kand617**
This PR is regarding #47 
All endpoints that return a list of objects now follow:
```
{
    items:[],
    totalCount:5
}
```

The total count could be null if total count is not retrievable. 
We do not have such as of now.

### CODE REVIEW

File: src/main/java/io/apicollab/server/dto/CollectionWrapperDTO.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+public class CollectionWrapperDTO<T> {
+    private static final long serialVersionUID = 7937487939121387950L;
+    private Integer totalCount;
+    private Collection<T> items;
+
+    public CollectionWrapperDTO(Collection<T> items, Integer totalCount) {
+        this.totalCount = totalCount;
+        this.items = items;
+    }
+
```
Comment (sudhirtumati) [inline src/main/java/io/apicollab/server/dto/CollectionWrapperDTO.java]: Add a constructor (for convenience) that takes `items` as argument and calculates `totalCount` as `items.size()` and change current usages to use this constructor instead

When a consumer knows that `items` is a subset, they can use the constructor with both arguments.

File: src/main/java/io/apicollab/server/dto/CollectionWrapperDTO.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.Collection;
+
+@Data
+public class CollectionWrapperDTO<T> {
+    private static final long serialVersionUID = 7937487939121387950L;
+    private Integer totalCount;
+    private Collection<T> items;
+
+    public CollectionWrapperDTO(Collection<T> items, Integer totalCount) {
```
Comment (sudhirtumati) [inline src/main/java/io/apicollab/server/dto/CollectionWrapperDTO.java]: Can be replaced with `@AllArgsConstructor`

Comment (kand617) [reply src/main/java/io/apicollab/server/dto/CollectionWrapperDTO.java]: done :)

File: src/main/java/io/apicollab/server/dto/CollectionWrapperDTO.java (hunk 1 above)
Comment (kand617) [reply src/main/java/io/apicollab/server/dto/CollectionWrapperDTO.java]: OK, done

File: src/main/java/io/apicollab/server/dto/CollectionWrapperDTO.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package io.apicollab.server.dto;
+
+import lombok.Data;
+
+import java.util.Collection;
+
+@Data
+public class CollectionWrapperDTO<T> {
+    private static final long serialVersionUID = 7937487939121387950L;
```
[the remark this answers is not in the document]
Comment (kand617) [reply src/main/java/io/apicollab/server/dto/CollectionWrapperDTO.java]: done

**Pull request merged**